REFRAME.md

```diff
@@ -280,6 +282,45 @@ Response:
 {"PutIPNSResponse : {}"}
 ```
 
+#### GetPeerAddresses
+
+A message for getting the addresses of a libp2p peer
+
+```ipldsch
+    type GetPeerAddressesRequest struct {
+        ID Bytes # libp2p PeerID
+        RecordTypes [String]
+    }
+
+    type Multiaddresses [Bytes] # Each element in the list is the binary representation of a complete multiaddr without a peerID suffix
+    type Libp2pSignedPeerRecord Bytes # https://github.com/libp2p/specs/blob/8c967a22bfbaff1ae41072b358fdba7c5883b6a4/RFC/0003-routing-records.md
+
+    type PeerRecordType union {
+     | Multiaddresses "multiaddrs"
+     | Libp2pSignedPeerRecord "769" // the libp2p signed peer record entry interpreted as decimal https://github.com/multiformats/multicodec/blob/f5dd49f35b26b447aa380e9a491f195fd49d912c/table.csv#L129
+     | Any default
+    } representation keyed
+    
+    type GetPeerAddressesResponse struct {
+        Records [PeerRecordType]
+    }
+```
+
+##### DAG-JSON Examples
+
+Request: // TODO
+```
+{"GetPeerAddressesRequest" : {
+    "ID" : {"/":{"bytes":"AXIUBPnagss"}},
+    "Record" : {"/":{"bytes":"AXIUBPnagss"}}

```

Comment:
nit: there presumably isn't a record in the request, this would be in the response

Reply:
Yep, I didn't update any of the JSON examples yet (marked with TODO)